exp/etcdrestore/webhooks/rke2config.go

```diff
@@ -45,6 +66,431 @@ func (r *RKE2ConfigWebhook) SetupWebhookWithManager(mgr ctrl.Manager) error {
 }
 
 // Default implements webhook.Defaulter so a webhook will be registered for the type.
-func (r *RKE2ConfigWebhook) Default(_ context.Context, _ runtime.Object) error {
+func (r *RKE2ConfigWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	logger := log.FromContext(ctx)
+
+	logger.Info("Configuring system agent on for RKE2Config")
+
+	rke2Config, ok := obj.(*bootstrapv1.RKE2Config)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a RKE2Config but got a %T", obj))
+	}
+
+	planSecretName := strings.Join([]string{rke2Config.Name, "rke2config", "plan"}, "-")
+
+	if err := r.createSecretPlanResources(ctx, planSecretName, rke2Config); err != nil {
+		return apierrors.NewBadRequest(fmt.Sprintf("failed to create secret plan resources: %s", err))
+	}
+
+	serviceAccountToken, err := r.EnsureServiceAccountSecretPopulated(ctx, planSecretName)
+	if err != nil {
+		return apierrors.NewBadRequest(fmt.Sprintf("failed to ensure service account secret is populated: %s", err))
+	}
+
+	logger.Info("Service account secret is populated")
+
+	serverUrlSetting := &unstructured.Unstructured{}
```

Comment:
To make it importable we need a new go.mod file and dedicated file for this kind (=Setting)? I am not sure if that's worth the effort. I noticed the unstructured usage in few places, i.e https://github.com/rancher/turtles/blob/main/test/framework/rancher_helpers.go#L219-L221

Reply:
I believe we need to move the webhook pathing code to the internal package of its own, and if possible, move out the `internal/rancher` to and API level, because it is a public API. Then the use of the generated `Setting` as a typed resource will be easy and correct.

Reply:
In any case, that change seems to be out of scope of this PR and can be discussed separately IMO.